src/CoreWCF.WebHttp/src/CoreWCF/Description/WebHttpBehavior.cs

```diff
@@ -0,0 +1,1137 @@
+            }
+            else
+            {
+                if (operation.Messages.Count > 1)
+                {
+                    TryGetStreamParameterType(operation.Messages[1], operation, false, out dummy);
+                }
+            }
+        }
+
+        private IDispatchMessageFormatter GetDefaultDispatchFormatter(OperationDescription od, bool useJson, bool isWrapped)
+        {
+            DataContractSerializerOperationBehavior dcsob = ((KeyedByTypeCollection<IOperationBehavior>)od.OperationBehaviors).Find<DataContractSerializerOperationBehavior>();
+            if (useJson)
+            {
+                if (dcsob == null)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.JsonFormatRequiresDataContract, od.Name, od.DeclaringContract.Name, od.DeclaringContract.Namespace)));
+                }
+                return CreateDataContractJsonSerializerOperationFormatter(od, dcsob, isWrapped);
+            }
+            else
+            {
+                EndpointDispatcher dummyED = new EndpointDispatcher(new EndpointAddress("http://localhost/"), "name", "");
+                DispatchRuntime dispatchRuntime = dummyED.DispatchRuntime;
+                DispatchOperation dop = new DispatchOperation(dispatchRuntime, "dummyDispatch", "urn:dummy");
+                dop.Formatter = null;
+
+                if (dcsob != null)
+                {
+                    (dcsob as IOperationBehavior).ApplyDispatchBehavior(od, dop);
+                    return dop.Formatter;
+                }
+
+                XmlSerializerOperationBehavior xsob = ((KeyedByTypeCollection<IOperationBehavior>)od.OperationBehaviors).Find<XmlSerializerOperationBehavior>();
+                if (xsob != null)
+                {
+                    xsob = new XmlSerializerOperationBehavior(od, xsob.XmlSerializerFormatAttribute, _contractNamespace);
+                    (xsob as IOperationBehavior).ApplyDispatchBehavior(od, dop);
+                    return dop.Formatter;
+                }
+            }
+
+            return null;
+        }
+
+        internal virtual DataContractJsonSerializerOperationFormatter CreateDataContractJsonSerializerOperationFormatter(OperationDescription od, DataContractSerializerOperationBehavior dcsob, bool isWrapped)
+        {
+            //return new DataContractJsonSerializerOperationFormatter(od, dcsob.MaxItemsInObjectGraph, dcsob.IgnoreExtensionDataObject, isWrapped, false, JavascriptCallbackParameterName);
+
+            OperationDescription clonedOperation = new OperationDescription(od.Name, od.DeclaringContract);
+            foreach (MessageDescription message in od.Messages)
+            {
+                MessageDescription clonedMessage = message.Clone();
+                if (IsValidReturnValue(clonedMessage.Body.ReturnValue))
+                {
```

Comment:
This code isn't quite doing what was described [here](https://github.com/CoreWCF/CoreWCF/pull/548#discussion_r810344994). You need to do this:
```c#
if (IsValidReturnValue(clonedMessage.Body.ReturnValue))
{
    if (clonedMessage.Body.Parts.Count == 0)
    {
        if (clonedMessage.Body.ReturnValue.Type == typeof(Stream))
        {
            clonedMessage.Body.WrapperName = JsonGlobals.rootString;
            clonedMessage.Body.WrapperNamespace = string.Empty;
        }
    }
}
else
{
    if (clonedMessage.Body.Parts.Count == 1)
    {
        if (clonedMessage.Body.Parts[0].Type == typeof(Stream))
        {
            clonedMessage.Body.WrapperName = JsonGlobals.rootString;
            clonedMessage.Body.WrapperNamespace = string.Empty;
        }
    }
}

```
Basically this is checking does the message have a return value which is a stream, if so fix up the WrapperName and WrapperNamespace of the boxy. If it doesn't have a return value, check if there is exactly one message body part, and if there is and it's type is Stream, then fix up the wrapper name and namespace.

Reply:
This has been addressed.